docs/use_cases/recommender_systems.md

```diff
@@ -1,15 +1,84 @@
-2. or you can give an explanation on how it should be and we will help you create one. Please tag arunesh@superlinked.com or @AruneshSingh (GitHub) in this case. -->
+![](assets/use_cases/recommender_systems/dummy_nodes.jpg)
+
+Recommender Systems are becoming increasingly important, given the plethora of products offered to the users/customers. For example, in early days, fashion retailers developed basic versions of content-based recommenders that served quite well, but soon the need emerged for more relevant recommendations that could increase user’s engagement. At the moment of this writing, fashion retailers use far more sophisticated approaches that make use not only of the user’s purchasing/viewing history, but also of user and item metadata: user age, location, spending habits, mood, etc.; item category, popularity, etc.
+
+However, it is sometimes the case that only scarce side information about the users or the items is available to us. Consider, for example, a public service provider that has its own audio- and video-on-demand platform; such companies have restrictions with regard to collecting/storing user metadata. But even in this case, we still have useful information about the items that are served, such as genre, popularity, etc. Still, developers often disregard this side information, because it is scarce, and proceed with collaborative-filtering (CF) approaches, which make use of the historical interactions only (data consisting of user-item pairs). While CF work reasonably well in this use case - extrapolate user preferences via similarities of all-user’s browsing/purchasing history - one question remains: can we improve on the quality by (somewhow) making use of available information, regardless how scarce? More precisely, can we use the collaborative filtering paradigm in this case, with this scarce side info? Yes, there are libraries that allow us to ‘inject’ side information (LightFM, for example). But there is also a way to extend the most efficient and effective collaborative filtering models, such as ALS Matrix Factorization (MF) (the Python ‘implicit’ library) or EASE, to make use of the side information.
+
+
+### Recommender Systems as Graphs
+
+Before we go into details on how this is achieved (at you’d be surprised how easily!), let’s take a step back and try to observe the typical matrix factorization approach through the lens of graph algorithms. The user-item interaction matrix can be observed as an adjacency matrix of a bipartite graph (‘bi’, because there are two groups of nodes, users and items, and link can exist only between nodes belonging to different groups). The rating recovered (predicted) by the MF approach can be seen as the predicted weight of the edge existing between that particular user-item pair. So, in the end, the recommendations for a user are actually the most likely links this user is going to establish in the future (with the information on the weight of these links).
```

Comment:
"take step back and try to observe" - I suggest to try and simplify all such statements.

Basically, pretend you are explaining all this to me in person - "write like you talk", see http://www.paulgraham.com/talk.html